Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/task_list/TaskListActivity.java

```diff
@@ -1,37 +1,50 @@
 package pg.autyzm.friendly_plans.manager_app.view.task_list;
 
 import android.content.Intent;
+import android.databinding.DataBindingUtil;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import database.entities.TaskTemplate;
 import database.repository.TaskTemplateRepository;
+import java.util.ArrayList;
+import java.util.List;
 import javax.inject.Inject;
 import pg.autyzm.friendly_plans.ActivityProperties;
 import pg.autyzm.friendly_plans.App;
 import pg.autyzm.friendly_plans.R;
+import pg.autyzm.friendly_plans.databinding.ActivityTaskListBinding;
 import pg.autyzm.friendly_plans.manager_app.view.task_create.TaskCreateActivity;
 import pg.autyzm.friendly_plans.notifications.ToastUserNotifier;
 
-public class TaskListActivity extends AppCompatActivity {
+public class TaskListActivity extends AppCompatActivity implements TaskListActivityEvents {
 
     @Inject
     TaskTemplateRepository taskTemplateRepository;
     @Inject
     ToastUserNotifier toastUserNotifier;
 
     private TaskRecyclerViewAdapter taskListAdapter;
+    private List<TaskTemplate> taskItemList;
```

Comment:
For me it is complicated and I am not sure we need these 4 lists. One should be enough.  After clicking on each button new items fetched from db base on task type could be assigned. Please consider that change, it will simplify the code and help in further maintain.